### PR TITLE
Add FLAGS_CI_both_cpu_and_gpu

### PR DESCRIPTION
**Pull request by 0x3878f**
### PR Category

Others

### PR Types

Others

### Description

Add one flag named 'FLAGS_CI_both_cpu_and_gpu' to optimize coverage pipeline.
There are many CPU unit tests that are redundantly run in both the py3 and coverage pipelines. This FLAG is used to disable CPU unit tests in the coverage pipeline.

Total time of coverage CI before optimization: **3h40m**

Total time of coverage CI after optimization: **3h22m**

The total time of coverage CI was reduced by about **18 minutes**

pcard-67164

### CODE REVIEW

**Review by risemeup1 — DISMISSED** [review]
LGTM

**Review by wanghuancoder — DISMISSED** [review]

**Review by phlrain — DISMISSED** [review]

**Review by XieYunshen — DISMISSED** [review]
LGTM

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

File: test/legacy_test/test_unflatten.py [hunk 1]
```diff
@@ -69,7 +70,13 @@ def setUp(self):
         self.set_api()
         self.set_args()
         self.get_output()
-        self.places = [paddle.CPUPlace()]
+        self.places = []
+        if (
+            os.environ.get('FLAGS_CI_both_cpu_and_gpu', 'False').lower()
```
Comment (Aurelius84) [inline test/legacy_test/test_unflatten.py:75]: 这里建议封装成一个公共函数放到类似op_test.py里，屏蔽实现细节，后续升级改造时也不用一个个单测来改了

**Review by Aurelius84 — DISMISSED** [review]
LGTM overall

File: test/legacy_test/test_unflatten.py (hunk 1 above)
Comment (0x3878f) [reply test/legacy_test/test_unflatten.py:75]: 好的，目前是先合入代码，并在`test/legacy_test/utils.py`增加了公共函数`get_places`，后续可完成对相关代码的替换。后面计划再添加一个拦截策略，对新添加的单测进行拦截

**Review by risemeup1 — APPROVED** [review]

**Pull request merged**